pkg/apis/certmanager/validation/certificate_for_issuer.go

```diff
@@ -59,6 +59,14 @@ func ValidateCertificateForACMEIssuer(crt *v1alpha1.CertificateSpec, issuer *v1a
 		el = append(el, field.Invalid(specPath.Child("organization"), crt.Organization, "ACME does not support setting the organization name"))
 	}
 
+	if crt.Duration.Duration != 0 {
+		el = append(el, field.Invalid(specPath.Child("duration"), crt.Duration, "ACME does not support certificate durations"))
+	}
+
+	if crt.RenewBefore.Duration != 0 {
+		el = append(el, field.Invalid(specPath.Child("renewBefore"), crt.RenewBefore, "ACME does not support certificate renewal times"))
+	}
```

Comment:
How come we don't allow this to be configured with the ACME issuer? Happy to leave this as-is for now if there's a lot more consideration that needs to be made, but it seems like we could/should be able to allow this?

Reply:
I think it makes sense to remove this restriction, as many people may want to configure this for ACME.

We could put some kind of a 'sensible' limit for ACME certs though, based around the 90d validity that boulder offers. i.e. we won't allow the renewBefore to be >90d (and obviously the MinimumRenewBefore boundaries are still enforced by the generic code-path)

Reply:
If we remove this check, the only thing that will happen (I think) is that LE will ignore the notAfter field in the CSR and give back whatever it wants (90d). This behavior is already exhibited by the vault issuer if it is configured to only issue with a certain duration and we are test cases to verify it still works. It should just result in a schedule change log message. 

Renew before is no big deal to let people change (as long as they dont rate limit themselves). 

So in summary, we can get rid of both of these restrictions without issue I think